sui_types/src/messages.rs

```diff
@@ -311,35 +313,274 @@ where
     }
 }
 
-/// An transaction signed by a client. signature is applied on data.
+/// A transaction signed by a client, optionally signed by an authority (depending on `S`).
+/// `S` indicates the authority signing state. It can be either empty or signed.
+/// TODO: Fill comment.
 /// Any extension to Transaction should add fields to TransactionData, not Transaction.
-// TODO: this should maybe be called ClientSignedTransaction + SignedTransaction -> AuthoritySignedTransaction
-#[derive(Debug, Eq, Clone, Serialize, Deserialize)]
-pub struct Transaction {
+#[derive(Debug, Clone, Eq, Serialize, Deserialize)]
```

Comment:
`TransactionEnvelope` derives `Eq` but not `PartialEq`!?

Reply:
`TransactionEnvelope` could not derive `PartialEq` because we need to manually implement its `PartialEq` on the two instantiations.